tests/test_dummy.py

```diff
@@ -0,0 +1,240 @@
+
+
+def test_enumerate_anchor_groups():
+    mol = Chem.MolFromSmiles("C1CC2OOC12")
+    core = [0, 1, 2, 5]
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [3, 4])
+    assert_anchor_group_equality(groups_of_1, [[2], [5]])
+    assert_anchor_group_equality(groups_of_2, [[2, 1], [2, 5], [5, 0], [5, 2]])
+    assert_anchor_group_equality(groups_of_3, [[2, 1, 0], [2, 5, 0], [5, 0, 1], [5, 2, 1]])
+
+    # aspirin
+    mol = Chem.MolFromSmiles("CC(=O)OC1=CC=CC=C1C(=O)O")
+    core = [3, 4, 5, 6, 7, 8, 9, 10]
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [0, 1, 2, 11, 12])
+    assert_anchor_group_equality(groups_of_1, [[3], [10]])
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [0, 1, 2])
+    assert_anchor_group_equality(groups_of_1, [[3]])
+    assert_anchor_group_equality(groups_of_2, [[3, 4]])
+    assert_anchor_group_equality(groups_of_3, [[3, 4, 5], [3, 4, 9]])
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [11, 12])
+    assert_anchor_group_equality(groups_of_1, [[10]])
+    assert_anchor_group_equality(groups_of_2, [[10, 9]])
+    assert_anchor_group_equality(groups_of_3, [[10, 9, 8], [10, 9, 4]])
+
+
+def test_enumerate_allowed_dummy_ixns_3_anchors():
+
+    dummy_group = {0, 1, 5, 8}
+    anchor_group = [2, 3, 4]
+
+    allowed_ixns = enumerate_dummy_ixns(dummy_group, anchor_group)
+
+    for ixn in allowed_ixns:
+        assert len(ixn) == len(set(ixn))
+
+    assert (0, 1, 2, 3) in allowed_ixns
+    assert (0, 1, 2) in allowed_ixns
+    assert (1, 2) in allowed_ixns
+    assert (0, 2, 3) in allowed_ixns
+    assert (1, 0, 2, 3) in allowed_ixns
+    assert (5, 1, 2, 3) in allowed_ixns
+    assert (1, 5, 2) in allowed_ixns
+    assert (5, 0, 2) in allowed_ixns
+    assert (1, 5, 8) in allowed_ixns
+    assert (8, 5, 1) in allowed_ixns
+    assert (0, 8, 5, 1) in allowed_ixns
+    assert (5, 1, 0, 8) in allowed_ixns
+
+    assert (0, 1, 3) not in allowed_ixns
+    assert (0, 1, 4) not in allowed_ixns
+    assert (0, 2, 4) not in allowed_ixns
+    assert (0, 3) not in allowed_ixns
+    assert (0, 4) not in allowed_ixns
+    assert (5, 3, 4) not in allowed_ixns
+    assert (1, 5, 4) not in allowed_ixns
+
+
+def test_enumerate_allowed_dummy_ixns_2_anchors():
+
+    dummy_group = {0, 1}
+    anchor_group = [4, 5]
+
+    allowed_ixns = enumerate_dummy_ixns(dummy_group, anchor_group)
+
+    for ixn in allowed_ixns:
+        assert len(ixn) == len(set(ixn))
+
+    assert len(allowed_ixns) == 10
+
+    assert (0, 1) in allowed_ixns
+    assert (1, 0) in allowed_ixns
+    assert (0, 4) in allowed_ixns
+    assert (1, 4) in allowed_ixns
+    assert (0, 1, 4) in allowed_ixns
+    assert (1, 0, 4) in allowed_ixns
+    assert (0, 4, 5) in allowed_ixns
+    assert (1, 4, 5) in allowed_ixns
+    assert (0, 1, 4, 5) in allowed_ixns
+    assert (1, 0, 4, 5) in allowed_ixns
```

Comment:
> Style question: Is there a reason to prefer this way of specifying a set rather than something like

Mainly for consistency with the `not in` statements with the 3 anchor test - which would have to resort to something like `ref.intersection(b) == set()` or something. I can change this though if its too hard to read.

Reply:
> Documentation request: At minimum, a brief comment at the top of this test file about how the reference values were produced. Ideally, also a one-sentence description in each test of what is being asserted.
Will add

Reply:
> (Some of these catch my eye because only one of the patterns is included both forward and reverse ((0,1) and (1,0)), not sure if that's noteworthy...)

I should probably uniquify the bonds in the set based on our typical lexical ordering of the atom idxs